controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -223,7 +223,7 @@ public void getSegmentsBetween(Controller.StreamCutRange request, StreamObserver
         log.info("getSegmentsBetweenStreamCuts called for stream {} for cuts from {} to {}", request.getStreamInfo(), request.getFromMap(), request.getToMap());
         String scope = request.getStreamInfo().getScope();
         String stream = request.getStreamInfo().getStream();
-        authenticateExecuteAndProcessResults(() -> this.authHelper.checkAuthorization(
+        authenticateExecuteAndProcessResults(() -> this.authHelper.checkAuthorizationAndCreateToken(
```

Comment:
Some methods in this very same class invoke `checkAuthorization` (e.g., the method above with the typo in the name) while others invoke `check AuthorizationAndCreateToken`. I'm not sure what the criteria are for deciding what to call.

Reply:
The former checks for authorization only, while the latter sets a delegation token apart from checking for authorization. The delegation token is used subsequently by the client when talking to the segment store.

Reply:
I get that basic behavior @ravisharda , that's not the point. Why do we need to create a token for some calls and others we don't ? What's different in the call path that makes one require the creation of a token while the other didn't?

Clearly this distinction is what caused the bug because whoever wrote this code did not make the right assessment on whether a token needed to be created or not. I'm asking for some explicit criteria so that we can perhaps review the calls in this very same class and others and validate that they are doing the right thing.

Reply:
I completely agree with you. 

As of now, I don't have a complete rule-set that governs whether to set the delegation token or not. I'd need to discover those rules through experimentation and document them. I'd propose that we write up a new Wiki document that details out the auth design, beyond the very high-level design information covered under [this](https://github.com/pravega/pravega/wiki/PDP-23:-Pravega-security----encryption-and-Role-Based-Access-Control) PDP. In that new document, we can cover both the high-level design as well as important rules like these. 

We can cover this separately as part of a different issue if you agree.

Reply:
I am not sure if the fix provided here addresses all the API calls of batch client in its entirety (for e.g., the deprecated APIs of `BatchClientFactoryImpl` class)?

Reply:
@vijikarthi, there seems to be no deprecated API method in `BatchFactoryImpl` interface, There is indeed a deprecated API method in its implementation class `BatchClientFactoryImpl`. I'll verify it to check if works with auth enabled. Could you also specify if there are other batch client API methods that you suspect won't work with auth?